File: src/main/java/com/learn/todolist/application/UserAuthServiceImpl.java
package com.learn.todolist.application;

import com.learn.todolist.domain.User;
import com.learn.todolist.infra.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class UserAuthServiceImpl {
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;

    public void validateEmail(String email) {
        Boolean isExistEmail = userRepository.existsByEmail(email);
        if (isExistEmail) throw new IllegalArgumentException("Email is already exist");
    }

    public void validateNickName(String nickName) {
        Boolean isExistNickName = userRepository.existsByNickName(nickName);
        if (isExistNickName) throw new IllegalArgumentException("NickName is already exist");
    }

    public User validateLogin(String email, String password) {
        User user = userRepository.findByEmail(email).orElseThrow(() -> new IllegalArgumentException("User not found"));
        if (!passwordEncoder.matches(password, user.getPassword())) {
            throw new IllegalArgumentException("Password is not correct");
        }
        return user;
    }
}

File: src/main/kotlin/com/todolist/aplication/user/usecase/service/UserAuthServiceImpl.kt
package com.todolist.aplication.user.usecase.service

import com.todolist.aplication.commonRepository.UserRepository
import com.todolist.domain.user.entity.User
import com.todolist.domain.user.module.UserAuthService
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Component
import org.springframework.security.crypto.password.PasswordEncoder

private val logger = KotlinLogging.logger {}

@Component
class UserAuthServiceImpl(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder
): UserAuthService{
    override fun validateEmail(email: String): Boolean {
        logger.info { "Check Is Exist Email Check" }
        val isExistEmail = userRepository.existsByEmailAndDeletedAtIsNull(email)
        check(!isExistEmail) { "Email is already in use" }
        logger.info { "Unused Email Then Confirmed" }
        return isExistEmail
    }

    override fun validateNickName(nickName: String): Boolean {
        logger.info { "Check Is Exist NickName Check" }
        val isExistNickName = userRepository.existsByNickNameAndDeletedAtIsNull(nickName)
        check(!isExistNickName) { "NickName is already in use" }
        logger.info { "Unused NickName Then Confirmed" }
        return isExistNickName
    }

    override fun validateLogin(email: String, password: String): User {
        logger.info { "Check Is Exist User Check" }
        val user: User = userRepository.findByEmailAndDeletedAtIsNull(email) ?: throw UsernameNotFoundException("User Not Found")
        logger.info { "Exist User Then Confirmed" }
        logger.info { "Check Password Match && Check If Not Withdrawal User" }
        when {
            !passwordEncoder.matches(password, user.password) -> {
                throw IllegalArgumentException("Password Not Match")
            }
        }
        logger.info { "User Login Confirmed" }
        return user
    }
}